libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -4256,13 +4261,14 @@ void GCS_MAVLINK::send_sys_status()
     const AP_BattMonitor &battery = AP::battery();
     float battery_current;
     int8_t battery_remaining;
+    if (!battery.capacity_remaining_pct(battery_remaining)) {
+        battery_remaining = -1;
+    }
 
     if (battery.healthy() && battery.current_amps(battery_current)) {
-        battery_remaining = battery.capacity_remaining_pct();
         battery_current = constrain_float(battery_current * 100,-INT16_MAX,INT16_MAX);
     } else {
         battery_current = -1;
-        battery_remaining = -1;
```

Comment:
This looks like a change in behavior, `capacity_remaining_pct` doesn't account for if the battery is healthy or has current monitoring, so under this we could report a the current as -1 here, but still give a random value for the capacity.

Reply:
good catch!